dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -124,70 +156,90 @@ public static DependencyGraph getCompleteDependencies(Artifact artifact)
    */
   public static DependencyGraph getTransitiveDependencies(Artifact artifact)
       throws DependencyCollectionException, DependencyResolutionException {
-    
     // root node
-    DependencyNode node = resolveCompileTimeDependencies(artifact);  
+    DependencyNode node = resolveCompileTimeRootDependencies(artifact);
     DependencyGraph graph = new DependencyGraph();
-    preorder(new Stack<>(), node, graph);
-    
+    levelOrder(node, graph);
     return graph;
   }
-  
-  // TODO Dedup the next two methods. They are duplicate code with only one line difference.
-  // this finds the actual graph that Maven sees with no duplicates and at most one version per
-  // library.
-  @SuppressWarnings("unchecked")
-  private static void preorder(Stack<DependencyNode> path, DependencyNode current,
-      DependencyGraph graph) {
-    
-    path.push(current);
-    DependencyPath forPath = new DependencyPath();
-    for (DependencyNode node : path) {
-      if (!"system".equals(node.getDependency().getScope())) {
-        forPath.add(node.getArtifact());
-      }
-    }
-    graph.addPath(forPath);
-    
-    for (DependencyNode child : current.getChildren()) {
-      preorder((Stack<DependencyNode>) path.clone(), child, graph);
+
+  static DependencyGraph getTransitiveDependencies(List<Artifact> artifacts)
```

Comment:
This doesn't appear to be called anywhere, except for its test

Reply:
OK. We do need to watch out for adding code in advance of actual use though. Surprisingly often we end up not needing the function we think we do. Also, smaller PRs are easier to review.

Reply:
Some background why this PR grew bigger than initial plan:

- I wanted getTransitiveDependencies(List<Artifact> artifacts)
- Noticed that the DependencyGraph has DependencyPath sorted by pre-order
-- Maven's dependency resolution is level-order
- tree formatting in text requires pre-order

Reply:
Do we still need getTransitiveDependencies(List artifacts) ?

Reply:
The reason I'm pushing on this is that it seems like `resolveCompileTimeRootDependencies` is just the standard thing we want to do, and it seems like it should be named `resolveCompileTimeDependencies`. But, we have a separate `resolveCompileTimeDependencies`, which this function calls, but neither actually seem necessary yet.

Reply:
I also feel it's confusing. Let me revert the renaming of `resolveCompileTimeRootDependencies`.

There are 3 ways to build CollectionRequest. I wanted to separate the 1st case as `resolveCompileTimeRootDependencies`, and 3rd `resolveCompileTimeDependencies `

> There are three ways to create a dependency graph. First, only the root dependency can be given. Second, a root dependency and direct dependencies can be specified in which case the specified direct dependencies are merged with the direct dependencies retrieved from the artifact descriptor of the root dependency. And last, only direct dependencies can be specified in which case the root node of the resulting graph has no associated dependency.

https://maven.apache.org/resolver/apidocs/org/eclipse/aether/collection/CollectRequest.html